File: amway/lynx-ua-iac/terraform-ua/lambda/middleware/src/lynx-config/skype.go
package main

import (
	"bytes"
	"encoding/json"
	"net/http"
	"os"
)

const (
	SkypeURL    = "https://touch.epm-esp.projects.epam.com/bot-esp/message"
	ContentType = "application/json"
	SimpleType  = "simple"
)

type skypeMessage struct {
	Channel string `json:"channel"`
	Secret  string `json:"secret"`
	Type    string `json:"type"`
	Text    string `json:"text"`
}

func sendSkypeMessage(msg string) {
	sm := skypeMessage{Channel: os.Getenv("CHAT_ID"), Secret: os.Getenv("SKYPE_SECRET"), Type: SimpleType, Text: msg}
	p, _ := json.Marshal(sm)
	payload := bytes.NewReader(p)

	client := &http.Client{}
	req, err := http.NewRequest("POST", SkypeURL, payload)
	if err != nil {
		log.Errorf("%s %s \n", getReqID(), err)
		return
	}
	req.Header.Add("accept-version", "1.0.0")
	req.Header.Add("content-type", ContentType)
	resp, err := client.Do(req)
	if err != nil {
		log.Errorf("%s %s \n", getReqID(), err)
		return
	}
	resp.Body.Close()
}

File: amway/lynx-iac/terraform/lambda/create-pr-from-prod/src/skype.py
import requests


def send_skype_message(url, secret, channel, msg):
    payload = {
        "channel": channel,
        "secret": secret,
        "type": "simple",
        "text": msg
    }
    headers = {"Content-Type": "application/json", "accept-version": "1.0.0"}
    try:
        response = requests.post(url, headers=headers, json=payload)
    except Exception as e:
        print("Could not send skype meessage: {err}".format(err=e))
        return
    if response.status_code != 200:
        print("Skype webhook status code is not 200: {code} - {msg}".format(
            code=response.status_code, msg=response.text))
